src/holon/models/rule_actions/rule_action_add_set.py

```diff
@@ -279,3 +398,45 @@ def apply_action_to_queryset(self, filtered_queryset: QuerySet, value: str):
                         "is_rule_action_template": False,
                     },
                 )
+
+    def apply_to_scenario_aggregate(
+        self,
+        scenario_aggregate: ScenarioAggregate,
+        filtered_repository: RepositoryBaseClass,
+        value: str,
+    ) -> ScenarioAggregate:
+        """Set the number of filtered objects with the model specified in rule_action_add to value"""
+
+        # parse value
+        n = int(float(value))
+        if n < 0:
+            raise ValueError(f"Value to add cannot be smaller than 0. Given value: {n}")
+
+        # get parent type and foreign key field name
+
+        try:
+            parent_fk_field_name = next(
+                parent_fk_fieldname
+                for parent_type, parent_fk_fieldname in RuleActionUtils.get_parent_classes_and_field_names(
+                    self.model_to_add.__class__
+                )
+                if filtered_repository.base_model_type == parent_type
+            )
+        except:
+            raise ValueError(
+                f"Type {filtered_repository.base_model_type} in the filter does not match found parent type {RuleActionUtils.get_parent_classes_and_field_names(self.model_to_add.__class__)} for model type {self.model_to_add.__class__.__name__}"
+            )
+
+        # only take first n objects
+        for filtererd_object in filtered_repository:
+            for _ in range(n):
+                scenario_aggregate.add_object(
+                    self.model_to_add,
+                    self.model_to_add_base_class,
+                    {
+                        parent_fk_field_name: filtererd_object,
```

Comment:
For serialization we look at the xxx_id fields set with an id. Maybe we should update both here, so the whole object is up-to-date.
What do you think?